File: src/main/java/ai/devchat/idea/action/AddToDevChatAction.java
package ai.devchat.idea.action;

import ai.devchat.common.Log;
import ai.devchat.devchat.DevChatActionHandler;
import ai.devchat.devchat.handler.AddContextNotifyHandler;
import com.alibaba.fastjson.JSONObject;

public class AddToDevChatAction {
    private DevChatActionHandler actionHandler;

    public AddToDevChatAction() {
        actionHandler = DevChatActionHandler.getInstance();
    }

    public void execute(String filePath, String fileContent, String language, int startLine) {
        Log.info("Add to DevChat -> path: " + filePath +
                " content: " + fileContent +
                " language: " + language +
                " startLine: " + startLine);

        JSONObject payload = new JSONObject();
        payload.put("path", filePath);
        payload.put("content", fileContent);
        payload.put("languageId", language);
        payload.put("startLine", startLine);

        AddContextNotifyHandler addContextNotifyHandler = new AddContextNotifyHandler(actionHandler);
        addContextNotifyHandler.setPayload(payload);
        addContextNotifyHandler.executeAction();
    }
}

File: src/main/kotlin/ai/devchat/idea/action/AddToDevChatAction.kt
package ai.devchat.idea.action

import ai.devchat.common.Log.info
import ai.devchat.devchat.DevChatActions
import ai.devchat.devchat.handler.AddContextNotifyHandler
import com.alibaba.fastjson.JSONObject

class AddToDevChatAction {
    fun execute(filePath: String, fileContent: String, language: String, startLine: Int, invokeLater: Boolean = false) {
        info(
            "Add to DevChat -> path: " + filePath +
                    " content: " + fileContent +
                    " language: " + language +
                    " startLine: " + startLine
        )
        val payload = JSONObject()
        payload["path"] = filePath
        payload["content"] = fileContent
        payload["languageId"] = language
        payload["startLine"] = startLine
        if (invokeLater) {
            cache = payload
        } else {
            AddContextNotifyHandler(DevChatActions.ADD_CONTEXT_NOTIFY,null, payload).executeAction()
        }
    }

    companion object {
        var cache: JSONObject? = null
    }
}
